### ISSUE TITLE
PrgComponent errors occurs after some idle time

### ISSUE THREAD

**Issue by rudy1976s** [CakeDC/search#174]
I have implemented Search plugin following your guide and it works as it should be as soon as I leave the application idle for a certain amount of time.
After there is the following error

````
Notice (8): Undefined index: ItemType [APP/Plugin/Search/Controller/Component/PrgComponent.php, line 358]
Warning (4096): Argument 1 passed to PrgComponent::serializeParams() must be of the type array, null given, called in /app/Plugin/Search/Controller/Component/PrgComponent.php on line 359 and defined [APP/Plugin/Search/Controller/Component/PrgComponent.php, line 237]
Warning (2): array_merge() [function.array-merge]: Argument #2 is not an array [APP/Plugin/Search/Controller/Component/PrgComponent.php, line 362]
Warning (4096): Argument 1 passed to PrgComponent::exclude() must be of the type array, null given, called in /app/Plugin/Search/Controller/Component/PrgComponent.php on line 363 and defined [APP/Plugin/Search/Controller/Component/PrgComponent.php, line 294]
Warning (2): Invalid argument supplied for foreach() [APP/Plugin/Search/Controller/Component/PrgComponent.php, line 296]
````

After a reload all worked correctly again.
What does this could depend on ?

Thanks in advance

Rudy

**Comment by rudy1976s** [comment]
I have tried and the problem no longer occurs.
But I need cache!

**Comment by antoniovassell** [comment]
@rudy1976s Maybe disable cache on that page or controller that needs the search. Also how long is "idle for a while"?

**Comment by rudy1976s** [comment]
I will investigate and let you know.
Thank you for your support
Rudy

**Comment by rudy1976s** [comment]
Hello
I have investigated and the issue seemed directly related to cache management.
This is a big issue because we need to use cache almost in every controller involving model acting as "Search".
I have found a workaround which seemed to solve the problem: I have added a check in function "validateSearch": it seems that after some time caches invalidates but the behavior still connected with it.
This is my modification:

public function validateSearch(Model $Model, $data = null) {
		if (!empty($data)) {
			$Model->set($data);
		}
		if( isset($Model->data[$Model->alias]) && sizeof($Model->data[$Model->alias]) ) :
			$keys = array_keys($Model->data[$Model->alias]);
			foreach ($keys as $key) {
				if (empty($Model->data[$Model->alias][$key])) {
					unset($Model->data[$Model->alias][$key]);
				}
			}
			return true;
		else:
			return false;
		endif;
	}

Maybe this is not clean or correct but under my part resolved the issue

Thanks again for your help

Rudy
